js/adapt-stateful-session.js

```diff
@@ -1,261 +1,266 @@
-      const latency = questionView.getLatency();
-      Adapt.offlineStorage.set('interaction', id, response, result, latency, responseType);
+    const courseState = SCORMSuspendData.deserialize(suspendData.c);
+    const courseComplete = courseState[0];
+    const assessmentPassed = courseState[1];
+    const trackingIdModels = Adapt.data.filter(model => model.get('_type') === this._trackingIdType && model.has('_trackingId'));
+    const trackingIds = trackingIdModels.map(model => model.get('_trackingId'));
+    if (!trackingIds.length) {
+      Adapt.log.info(`course._isComplete: ${courseComplete}, course._isAssessmentPassed: ${assessmentPassed}, ${this._trackingIdType} completion: no tracking ids found`);
+      return;
     }
+    const data = SCORMSuspendData.deserialize(suspendData.q);
+    const max = Math.max(...data.map(item => item[0][0]));
+    const shouldStoreResponses = (data[0].length === 3);
+    const completionString = data.reduce((markers, item) => {
+      const trackingId = item[0][0];
+      const isComplete = shouldStoreResponses ?
+        item[2][1][0] :
+        item[1][0];
+      const mark = isComplete ? '1' : '0';
+      markers[trackingId] = (markers[trackingId] === '-' || markers[trackingId] === '1') ?
+        mark :
+        '0';
+      return markers;
+    }, (new Array(max + 1).join('-').split(''))).join('');
+    Adapt.log.info(`course._isComplete: ${courseComplete}, course._isAssessmentPassed: ${assessmentPassed}, ${this._trackingIdType} completion: ${completionString}`);
+  }
 
-    onTrackingComplete(completionData) {
-      const config = Adapt.spoor.config;
-      this.saveSessionState();
-      let completionStatus = completionData.status.asLowerCase;
-      // The config allows the user to override the completion state.
-      switch (completionData.status) {
-        case COMPLETION_STATE.COMPLETED:
-        case COMPLETION_STATE.PASSED: {
-          if (!config || !config._reporting || !config._reporting._onTrackingCriteriaMet) {
-            Adapt.log.warn(`No value defined for '_onTrackingCriteriaMet', so defaulting to '${completionStatus}'`);
-          } else {
-            completionStatus = config._reporting._onTrackingCriteriaMet;
-          }
+  onLanguageChanged() {
+    // when the user switches language, we need to:
+    // - reattach the event listeners as the language change triggers a reload of
+    //   the json, which will create brand new collections
+    // - get and save a fresh copy of the session state. as the json has been reloaded,
+    //   the blocks completion data will be reset (the user is warned that this will
+    //   happen by the language picker extension)
+    // - check to see if the config requires that the lesson_status be reset to
+    //   'incomplete'
+    const config = Adapt.spoor.config;
+    this.removeEventListeners();
+    this.setupEventListeners();
+    this.saveSessionState();
+    if (config?._reporting?._resetStatusOnLanguageChange === false) return;
```

Comment:
Should probably be.
```suggestion
    if (config?._reporting?._resetStatusOnLanguageChange !== true) return;
```
I think this was a `=== true` because it should only act when explictly set to `true` it'll rarely be set explicitly to `false`